docs/roadmap/scaling-2019.md

```diff
@@ -0,0 +1,115 @@
+# 2019 Autoscaling Roadmap
+
+This is what we hope to accomplish in 2019.
+
+## 2019 Goals
+
+### Sub-Second Cold Start
+
+Serverless is only as good as the illusion it sustains. Throwing some code into a "serverless" framework, the expectation is that it will just be running when it needs to, with as many resources as necessary. Including zero. But to realize that magical threshold and maintain the illusion of serverless, the code must come back as if it was never gone. The exact latency requirement of "as if it was never gone" will vary from use-case to use-case. But generally less than one second is a good start.
+
+Right now cold-starts are between 10 and 15 seconds which is an order of magnitude too slow. The time is spent starting the pod, waiting for Envoy to start and telling all nodes how to reach the pod through the Kubernetes Service. Without the Istio mesh (just routing request to individual pods as they come up) still takes about 4 seconds.
+
+We've poked at this problem in 2018 ([#1297](https://github.com/knative/serving/issues/1297)) but haven't been able to make significant progress. This area requires some dedicated effort to:
+
+1. identify and programatically capture sources of cold-start latency at all levels of the stack ([#2495](https://github.com/knative/serving/issues/2495))
+2. chase down the low hanging fruit (e.g. [#2659](https://github.com/knative/serving/issues/2659))
+3. architect solutions to larger chunks of cold-start latency
+
+**Our goal is to achieve sub-second average cold-starts by the end of the year.**
+
+* POC: Greg Haynes (IBM)
+* Github: [Project 8](https://github.com/knative/serving/projects/8)
+
+### Overload Handling
+
+Knative Serving provides concurrency controls to limit the number of requests a container must handle simultaneously. Additionally, each pod has a queue for holding requests when the container concurrency limit has been reached. When the pod-level queue overflows, subsequent request are rejected with 503 "overload".
+
+This is desirable to protect the Pod from being overloaded. But in the aggregate the behavior is not ideal for situations when autoscaling needs some time to react to sudden increases in request load.  This could be when the revision is scaled to zero.  Or when the revision is already running some pods, but not nearly enough.
+
+The goal of Overload Handling is to enqueue requests at a revision-level. Scale-from-zero should not overload if autoscaling can react in a reasonable amount of time to provide additional pods. When new pods come online, they should be able to take load for the existing pods. Even when scaled above zero, brief spikes of overload should be handled by enqueuing requests at a revision-level. The depth of the revision-level queue should also be configurable because even the Revision as a whole needs to guard against overload.
```

Comment:
> Scale-from-zero should not overload if autoscaling can react in a reasonable amount of time to provide additional pods.

Should this be an explicit test similar to the one below?

Reply:
I've actually removed the explicit tests in the roadmap.  That's too much detail I think.  I'll leave that to individual issues in the projects.